components/lifter.py

```diff
@@ -1,63 +1,99 @@
 import ctre
+import wpilib
 
 
 class Lifter:
-    lift_motor: ctre.WPI_TalonSRX
+    motor: ctre.WPI_TalonSRX
+    center_switch: wpilib.DigitalInput
+    top_switch: wpilib.DigitalInput
+
     #  TODO find encoder values, tune height values to include robot height and cube
-    COUNTS_PER_REV = 1
-    DISTANCE_PER_COUNT = 1  # in m
+    COUNTS_PER_REV = 4096
+    NUM_TEETH = 42
+    TEETH_DISTANCE = 0.005
+    COUNTS_PER_METER = COUNTS_PER_REV / (NUM_TEETH * TEETH_DISTANCE) / 2  # for height of dolly
+
+    MOTOR_FREE_SPEED = 5310 / 600  # RPM to rotations/100ms
+    GEAR_RATIO = 1 / 6
+    FREE_SPEED = int(MOTOR_FREE_SPEED * GEAR_RATIO * COUNTS_PER_REV)  # counts/100ms
+
+    TOP_HEIGHT = int(COUNTS_PER_METER * 0.5)  # in counts
+    BOTTOM_HEIGHT = int(COUNTS_PER_METER * 0)
+    THRESHOLD = 10
 
-    SWITCH_HEIGHT = 47.625
-    LOWER_SCALE = 121.92
-    BALANCED_SCALE = 152.4
-    UPPER_SCALE = 182.88
-    GROUND_HEIGHT = 0
+    HEIGHT_FROM_FLOOR = 0.2585  # height from floor to initial lift pos when reset in m
+    CONTAINMENT_SIZE = 0  # height needed for the mechanism to work properly in m
 
-    THRESHOLD = 100
+    UPPER_SCALE = 1.8288 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE  # in m
+    BALANCED_SCALE = 1.524 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
+    LOWER_SCALE = 1.2192 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
+    SWITCH_HEIGHT = 0.47625 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
 
     def setup(self):
         """This is called after variables are injected by magicbot."""
-        self.set_pos = self.GROUND_HEIGHT
+        self.set_pos = None
```

Comment:
This whole file would be much more readable if `set_pos` was in meters, not counts, and you did the conversion to encoder counts in `execute` and `get_pos`